.teamcity/common/performance-test-extensions.kt

```diff
@@ -55,8 +58,37 @@ fun distributedPerformanceTestParameters(workerId: String = "Gradle_Check_Indivi
     "-Porg.gradle.performance.buildTypeId=$workerId -Porg.gradle.performance.workerTestTaskName=fullPerformanceTest -Porg.gradle.performance.coordinatorBuildId=%teamcity.build.id%"
 )
 
-val individualPerformanceTestArtifactRules = """
-        subprojects/*/build/test-results-*.zip => results
-        subprojects/*/build/tmp/**/log.txt => failure-logs
-        subprojects/*/build/tmp/**/profile.log => failure-logs
-    """.trimIndent()
+const val individualPerformanceTestArtifactRules = """
+subprojects/*/build/test-results-*.zip => results
+subprojects/*/build/tmp/**/log.txt => failure-logs
+subprojects/*/build/tmp/**/profile.log => failure-logs
+"""
+
+fun BuildSteps.killGradleProcessesStep(os: Os) {
+    script {
+        name = "KILL_GRADLE_PROCESSES"
+        executionMode = BuildStep.ExecutionMode.ALWAYS
+        scriptContent = os.killAllGradleProcesses
+    }
+}
+
+// to avoid pathname too long error
+fun BuildSteps.windowsSubstrStep(os: Os) {
+    if (os == Os.WINDOWS) {
+        script {
+            name = "SETUP_VIRTUAL_DISK_FOR_PERF_TEST"
+            executionMode = BuildStep.ExecutionMode.ALWAYS
+            scriptContent = """subst p: "%teamcity.build.checkoutDir%" """
+        }
+    }
+}
+
+fun BuildSteps.removeWindowsSubstrStep(os: Os) {
```

Comment:
```suggestion
fun BuildSteps.removeWindowsSubstStep(os: Os) {
```

This is `subst` not `substr`.